reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/WindowSized.kt

```diff
@@ -0,0 +1,96 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.base.setCancellable
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.disposable.DisposableWrapper
+import com.badoo.reaktive.subject.unicast.UnicastSubject
+import com.badoo.reaktive.utils.atomic.AtomicBoolean
+import com.badoo.reaktive.utils.atomic.AtomicInt
+import com.badoo.reaktive.utils.atomic.AtomicLong
+import com.badoo.reaktive.utils.queue.SharedQueue
+
+/**
+ * Please refer to the corresponding RxJava
+ * [document](http://reactivex.io/RxJava/javadoc/io/reactivex/Observable.html#window-long-long-).
+ */
+fun <T> Observable<T>.window(
+    count: Long,
+    skip: Long = count
+): Observable<Observable<T>> {
+    require(count > 0) { "count > 0 required but it was $count" }
+    require(skip > 0) { "skip > 0 required but it was $skip" }
+
+    return observable { emitter ->
+        val upstreamDisposableWrapper = DisposableWrapper()
+
+        val windows = SharedQueue<UnicastSubject<T>>()
+        val activeWindowsCount = AtomicInt(1)
+
+        emitter.setCancellable {
+            if (activeWindowsCount.addAndGet(-1) == 0) {
+                upstreamDisposableWrapper.dispose()
+            }
+        }
+
+        val onTerminate: () -> Unit = {
+            if (activeWindowsCount.addAndGet(-1) == 0 && emitter.isDisposed) {
+                upstreamDisposableWrapper.dispose()
+            }
+        }
+
+        subscribe(object : ObservableObserver<T> {
+
+            private val skippedCount = AtomicLong()
+            private val tailWindowValuesCount = AtomicLong()
+
+            override fun onSubscribe(disposable: Disposable) {
+                upstreamDisposableWrapper.set(disposable)
+            }
+
+            override fun onNext(value: T) {
+                val skipped = skippedCount.value
+
+                val window: UnicastSubject<T>?
+                val windowSubscribed: AtomicBoolean?
```

Comment:
We could introduce a separate private wrapper class, same as in RxJava and as I did here: #535 
In this case you will have only variable to check for null. Also this method will be cleaner.

Reply:
Done.